src/braket_container.py

```diff
@@ -254,41 +226,84 @@ def install_additional_requirements() -> None:
+def in_extracted_code_dir():
+    current_dir = os.getcwd()
+    try:
+        os.chdir(EXTRACTED_CUSTOMER_CODE_PATH)
+        yield
+    finally:
+        os.chdir(current_dir)
+
+
+def wrap_customer_code(customer_method: Callable) -> Callable:
+    def wrapped_customer_code(**kwargs):
+        try:
+            with in_extracted_code_dir():
+                return customer_method(**kwargs)
+        except Exception as e:
+            exception_type = type(e).__name__
+            exception_string = (
+                exception_type
+                if not str(e)
+                else f"{exception_type}: {e}"
+            )
+            _log_failure(exception_string, display=False)
+            raise e
+    return wrapped_customer_code
+
+
+def kick_off_customer_script(customer_code: Callable) -> multiprocessing.Process:
     """
-    When provided the name of the package and the method to run, we run them as a process.
+    Runs the customer script as a separate process.
 
     Args:
-        entry_point (str): the code to run in the format <package>:<method>.
+        customer_code (Callable): The customer method to be run.
 
     Returns:
-        int: The exit code of the customer code run.
+        Process: the process handle to the running process.
     """
     print("Running Code As Process")
-    customer_code_process = kick_off_customer_script(entry_point)
-    join_customer_script(customer_code_process)
-    print("Code Run Finished")
-    return customer_code_process.exitcode
+    wrapped_customer_code = wrap_customer_code(customer_code)
+    process_kwargs = {"target": wrapped_customer_code}
+
+    function_args = try_bind_hyperparameters_to_customer_method(customer_code)
+    if function_args is not None:
+        process_kwargs["kwargs"] = function_args
 
+    customer_code_process = multiprocessing.Process(**process_kwargs)
+    customer_code_process.start()
+    return customer_code_process
 
-def run_customer_code_as_subprocess(entry_point: str) -> int:
+
+def join_customer_script(customer_code_process: multiprocessing.Process):
     """
-    When provided just the name of the module to run, we run it as a subprocess.
+    Joins the process running the customer code.
 
     Args:
-        entry_point (str): the name of the module to run.
-
-    Returns:
-        int: The exit code of the customer code run.
+        customer_code_process (Process): the process running the customer code.
     """
-    print("Running Code As Subprocess")
     try:
-        result = subprocess.run(["python", "-m", entry_point], cwd=EXTRACTED_CUSTOMER_CODE_PATH)
+        customer_code_process.join()
     except Exception as e:
-        log_failure_and_exit(f"Unable to run job at entry point {entry_point}\nException: {e}")
+        log_failure_and_exit(f"Job did not exit gracefully.\nException: {e}")
     print("Code Run Finished")
-    return_code = result.returncode
-    return return_code
+    return customer_code_process.exitcode
```

Comment:
You're actually not using this returned value.

Reply:
updated